src/sentry/web/frontend/oauth_authorize.py

```diff
@@ -127,17 +127,25 @@ def get(self, request: HttpRequest, **kwargs) -> HttpResponseBase:
                 err_response="client_id",
             )
 
-        # TODO (athena): Clean up this so scopes are always coming from the model
-        # This change is temporarily needed before we migrate existing applications
-        # to have the correct scopes
-        if application.requires_org_level_access:
-            scopes = application.scopes
+        scopes = request.GET.get("scope")
+        if scopes:
+            scopes = scopes.split(" ")
         else:
-            scopes = request.GET.get("scope")
-            if scopes:
-                scopes = scopes.split(" ")
-            else:
-                scopes = []
+            scopes = []
+        if application.requires_org_level_access:
+            # Applications that require org level access have a maximum scope limit set
+            # in admin that should not pass
+            max_scopes = application.scopes
```

Comment:
i couldn't find where this is set, would `max_scopes` include all the "lower" scopes as well?
example: it would include read if we write was granted?

or do we explicitly grant each scope?

Reply:
That's a good question. Explicit grant is not required. For example if I select event:admin in the UI the model stores: `scopes=["event:read","event:write","event:admin"]`

Reply:
It happens at the UI level: https://github.com/getsentry/sentry/blob/14dfa8a7b3138e32ed9f03add0d0b93fe60bfebb/static/app/constants/index.tsx#L139

Which is why in the unit tests I can't re-create this situation. At API level you have to mention explicitly.